packages/web-components/src/calendar/calendar.ts

```diff
@@ -0,0 +1,368 @@
+    const today: Date = new Date();
+
+    //when the month picker is not visible, this.monthPickerYear is always going to be the current year.
+    return this.month === today.getMonth() + 1 &&
+      this.year === today.getFullYear() &&
+      this.monthPickerYear === today.getFullYear()
+      ? 'slotted-link inactive'
+      : 'slotted-link';
+  }
+
+  /**
+   * Creates a class string for cells on the right panel
+   * @returns - string of class names
+   * @public
+   */
+  public getRightCellClassNames(detail: number, todayMonth: number, todayYear: number) {
+    const isToday = this.yearPickerOpen ? detail === todayYear : detail === todayMonth;
+
+    return ['right-cell-outer', this.hasAttribute('highlightCurrentMonth') && isToday && 'right-panel-today']
+      .filter(Boolean)
+      .join(' ');
+  }
+
+  /**
+   * Returns a list of month labels
+   * @returns A 2D array of month texts
+   * @public
+   */
+  public getMonthText(): { text: string; detail: number }[][] {
+    const months = this.dateFormatter.getMonths();
+    const monthsText: { text: string; detail: number }[][] = [];
+    let monthCount = 0;
+
+    while (monthCount < months.length || monthsText[monthsText.length - 1].length % 4 !== 0) {
+      const month = { text: months[monthCount], detail: monthCount + 1 };
+      const target = monthsText[monthsText.length - 1];
+      if (monthsText.length === 0 || target.length % 4 === 0) {
+        monthsText.push([month]);
+      } else {
+        target.push(month);
+      }
+      monthCount++;
+    }
+    return monthsText;
+  }
+
+  /**
+   * Returns a list of year labels for a decade
+   * @returns A 2D array of year texts
+   * @public
+   */
+  public getDecadeText(decadeStartYear: number): { text: string; detail: number }[][] {
+    const decade = this.dateFormatter.getDecade(decadeStartYear);
+    const decadeText: { text: string; detail: number }[][] = [];
+    let yearCount = 0;
+
+    while (yearCount < decade.length || decadeText[decadeText.length - 1].length % 4 !== 0) {
+      const yearText = { text: decade[yearCount], detail: decadeStartYear + yearCount };
+      const target = decadeText[decadeText.length - 1];
+      if (decadeText.length === 0 || target.length % 4 === 0) {
+        decadeText.push([yearText]);
+      } else {
+        target.push(yearText);
+      }
+      yearCount++;
+    }
+    return decadeText;
+  }
+
+  /**
+   * Updates calendar to show today when user clicks on "Go to today"
+   * @param event - mouse event for clicking on the link
+   */
+  public handleGoToToday(event: MouseEvent) {
+    const today: Date = new Date();
+    this.switchMonth(today.getMonth() + 1, today.getFullYear());
+    this.yearPickerOpen = false;
+  }
+
+  /**
+   * Emits the "rightcellselected" event with the seleced month.
+   * @param cellDetail - The month of year that's been selected
+   * @public
+   */
+  public handleRightCellSelect(event: Event, cellDetail: number): void {
```

Comment:
If we're not doing any processing of the event here, we can remove this method and emit the `rightCellSelected` event when clicked. Change [this code](https://github.com/microsoft/fluentui/pull/28585/files#diff-920368126b494bd936b5d4fe876e865fafc56efe6ccb29c93ba97a5008184d5eR49-R62:~:text=%3C%24%7BcellTag,cellTag%7D%3E) to:

```typescript
       <${cellTag}
          class="${(x, c) => c.parentContext.parent.getRightCellClassNames(x.detail, todayMonth, todayYear)}"
          part="month-cell"
          tabindex="-1"
          role="gridcell"
          grid-column="${(x, c) => c.index + 1}"
          @click="$emit('rightCellSelected', $event)"
      >
        <div
        class="right-cell">
          ${x => x.text}
        </div>
        </slot></slot>
      </${cellTag}>

```

Reply:
This has since been revised by Michelle to emit the corresponding event.